metaflow/metaflow_current.py

```diff
@@ -62,7 +62,12 @@ def _set_env(
 
     def _update_env(self, env):
         for k, v in env.items():
-            setattr(self.__class__, k, property(fget=lambda _, v=v: v))
+            if isinstance(v, property):
+                setattr(self.__class__, k, v)
```

Comment:
nit: I would make sure that the property does not have a setter. I know with the callable, it could technically be broken too but still may be a good idea to enforce it a bit.